src/CoordinateSystemsCore/CoordinateSystemsCore.jl

```diff
@@ -0,0 +1,33 @@
+module CoordinateSystemsCore
+
+using StaticArrays
+
+export AbstractCoordinateSystem, Cartesian, Spherical
+export point, coordinate, coordinate_system, with_coordinate_system
+export AbstractPoint, Point
+
+# We accept several user input types, but always store and compute as SVector
+const CoordinateType{N} = Union{NTuple{N}, SVector{N}}
```

Comment:
I would not fall back to SVector or NTuple because that is gonna cause the same issues we have currently with SVector everywhere. We need to really enforce some Cartesian type to avoid SVector proliferation.

Reply:
This would allow users to create points in a convenient way, e.g., `point(2, 3)`. But internally it's still stored and computed as `SVector`.